src/components/ContactsAdd.js

```diff
@@ -1,39 +1,153 @@
+    Twitter:""
+  })
+  
+  
   //TODO: Implement controlled form
   //send POST to json server on form submit
+  const navigate = useNavigate();
+
+  const handleSubmit = (event) => {
+   
+    const submit = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(contact),
+    };
+    fetch("http://localhost:4000/contacts", submit)
+      .then(setContacts([...contacts,contact ]))
+      .then(navigate('/contact'));
+    //  setContacts(contact)
+    event.preventDefault();
+    
+  };
+
+
+  const handleChange = (event) => {
+    const inputValue = event.target.value;
+    const inputName = event.target.name;
+
+    console.log("hadleChange", inputValue, inputName);
+    if (inputName === "firstName") {
+      setContact({ ...contact, firstName: inputValue });
+    }
+    if (inputName === "lastName") {
+      setContact({ ...contact, lastName: inputValue });
+    }
+    if (inputName === "lastName") {
+      setContact({ ...contact, lastName: inputValue });
+    }
+    if (inputName === "street") {
+      setContact({ ...contact, street: inputValue });
+    }
+    if (inputName === "city") {
+      setContact({ ...contact, city: inputValue });
+    }
+    if (inputName === "LinkedIn") {
+      setContact({ ...contact, LinkedIn: inputValue });
+    } if (inputName === "Email") {
+      setContact({ ...contact, Email: inputValue });
+    } if (inputName === "Twitter") {
+      setContact({ ...contact, Twitter: inputValue });
+    }
+  };
 
   return (
-    <form className="form-stack contact-form">
+    <form className="form-stack contact-form" onSubmit={handleSubmit}>
       <h2>Create Contact</h2>
 
       <label htmlFor="firstName">First Name</label>
-      <input id="firstName" name="firstName" type="text" required />
+      <input
+        id="firstName"
+        name="firstName"
+        type="text"
+        value={contact.firstName}
+        onChange={handleChange}
+        required
+      />
 
       <label htmlFor="lastName">Last Name:</label>
-      <input id="lastName" name="lastName" type="text" required/>
+      <input
+        id="lastName"
+        name="lastName"
+        type="text"
+        value={contact.lastName}
+        onChange={handleChange}
+        required
+      />
 
       <label htmlFor="street">Street:</label>
-      <input id="street" name="street" type="text" required/>
+      <input
+        id="street"
+        name="street"
+        type="text"
+        value={contact.street}
+        onChange={handleChange}
+        required
+      />
 
       <label htmlFor="city">City:</label>
-      <input id="city" name="city" type="text" required/>
+      <input
+        id="city"
+        name="city"
+        type="text"
+        value={contact.city}
+        onChange={handleChange}
+        required
+      />
+        <label htmlFor="Email">Email:</label>
+      <input
+        id="Email"
+        name="Email"
+        type="text"
```

Comment:
type should be email rather than just text, this will make sure that if required is set that the user has to input an email instead of just any string